packages/headless/src/features/commerce/common/actions.ts

```diff
@@ -22,6 +23,7 @@ import {SortBy, SortCriterion} from '../sort/sort';
 
 export type StateNeededByQueryCommerceAPI = ConfigurationSection &
   ProductListingV2Section &
+  Partial<RecommendationsSection> &
```

Comment:
We should extend the state like we do for search instead: https://github.com/coveo/ui-kit/blob/8f4d68d7a54d6aa539ed85b7fd0516c123097eb7/packages/headless/src/features/commerce/search/search-actions.ts#L15-L16
```suggestion
```

Reply:
I've made the change but not 100% this is the correct solution